server/accesslog.go

```diff
@@ -126,10 +126,17 @@ func (a *AccessLogServer) handleHTTPLogs(msg *als.StreamAccessLogsMessage_HttpLo
 
 		var api string
 		var authContext *auth.Context
+		var grpcService string
+		var operation string
 
 		extAuthzMetadata := getMetadata(extAuthzFilterNamespace)
 		if extAuthzMetadata != nil {
-			api, authContext = a.handler.decodeAuthMetadata(extAuthzMetadata.GetFields())
+			metadata, err := a.handler.decodeAuthMetadata(extAuthzMetadata.GetFields())
+			if err != nil {
+				log.Debugf(err.Error())
+				continue
+			}
+			api, authContext, grpcService, operation = metadata.Api, metadata.Ac, metadata.GrpcService, metadata.Operation
```

Comment:
Although Go allows parallel assignment like this, it isn't idiomatic. Generally, you'd do them individually, or create some sort of struct type that would contain the results, and build it inline, here. I'd suggest taking a look and see if you really need these as individual fields, or if you can just operate on `metadata` directly for the rest of this func.

Reply:
After looking at how the rest of the code is used, I found that `func (h *Handler) decodeMetadataHeaders(headers map[string]string) (string, *auth.Context)` followed the same return structure that I refactored in `decodeAuthMetadata`.  Having the api and authContext as what is returned causes these values to need to be individual fields.

Similar to `decodeMetadataHeaders`, I have updated this PR so `decodeMetadataHeaders` now returns `(*AuthMetadata, error)`.  This refactor causes both methods to follow the same pattern for decoding, so I was able to refactor out the use of individual fields.